src/main/java/org/simplify4u/plugins/sign/openpgp/PGPSecretKeyUtils.java

```diff
@@ -133,4 +135,30 @@ private static String fingerprintToString(byte[] bytes) {
         }
         return ret.toString();
     }
+
+    /**
+     * Verify expiration time of secret key.
+     *
+     * @param secretKey     a key to check
+     * @param secretKeyRing a keyRing used for prepare message
+     *
+     * @throws PGPSignerException if key expired
+     */
+    public static void verifyKeyExpiration(PGPSecretKey secretKey, PGPSecretKeyRing secretKeyRing) {
+
+        long validSeconds = secretKey.getPublicKey().getValidSeconds();
+        if (validSeconds > 0) {
+
+            LocalDateTime expireDateTime = secretKey.getPublicKey().getCreationTime()
```

Comment:
Sad but true, bouncy castle provides only local time but no time zone, so if someone ships the key into a different zone it effectively works up to 24 hours longer or shorter than actually expected.